src/bin/pgcopydb/extensions.c

```diff
@@ -405,3 +405,122 @@ copydb_parse_extensions_requirements(CopyDataSpec *copySpecs, char *filename)
 
 	return true;
 }
+
+
+/*
+ * copydb_prepare_extensions_restore implements pre pg_restore steps that might
+ * be needed for some extensions.
+ *
+ * At the moment we need to call timescaledb_pre_restore() when timescaledb has
+ * been used.
+ */
+bool
+copydb_prepare_extensions_restore(CopyDataSpec *copySpecs)
+{
+	bool hasTimescaledb = false;
+	DatabaseCatalog *filterDB = &(copySpecs->catalogs.filter);
+	const char *extensionName = "timescaledb";
+
+	if (!catalog_extension_exists(filterDB, extensionName, &hasTimescaledb))
+	{
+		/* errors have already been logged*/
+		return false;
+	}
+    
+	if (hasTimescaledb)
+	{
+		log_debug("Timescaledb extension is present");
+		if (!timescaledb_pre_restore(copySpecs))
+		{
+			/* errors have already been logged */
+			return false;
+		}
+	}
+	return true;
+}
+
+
+/*
+ * copydb_prepare_extensions_restore implements pre pg_restore steps that might
+ * be needed for some extensions.
+ *
+ * At the moment we need to call timescaledb_pre_restore() when timescaledb has
+ * been used.
+ */
+bool
+copydb_finalize_extensions_restore(CopyDataSpec *copySpecs)
+{
+	bool hasTimescaledb = false;
+	DatabaseCatalog *filterDB = &(copySpecs->catalogs.filter);
+	const char *extensionName = "timescaledb";
+
+    if (!catalog_extension_exists(filterDB, extensionName, &hasTimescaledb))
+	{
+		/* errors have already been logged*/
+		return false;
+	}
+
+	if (hasTimescaledb)
+	{
+		log_debug("Timescaledb extension is present");
+		if (!timescaledb_post_restore(copySpecs))
+		{
+			/* errors have already been logged */
+			return false;
+		}
+	}
+
+	return true;
+}
+
+
+/*
+ * Call the timescaledb_pre_restore() SQL function on the target database.
+ */
+bool
+timescaledb_pre_restore(CopyDataSpec *copySpecs)
+{
+	PGSQL dst = { 0 };
+
+	if (!pgsql_init(&dst, copySpecs->connStrings.target_pguri, PGSQL_CONN_TARGET))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	char *sql = "SELECT timescaledb_pre_restore()";
+
+	if (!pgsql_execute(&dst, sql))
+	{
+		log_error("Failed to call timescaledb_pre_restore()");
+		return false;
+	}
+
+	return true;
+}
+
+
+/*
+ * Call the timescaledb_post_restore() SQL function on the target database.
+ */
+bool
+timescaledb_post_restore(CopyDataSpec *copySpecs)
+{
+	PGSQL dst = { 0 };
+
+	if (!pgsql_init(&dst, copySpecs->connStrings.target_pguri, PGSQL_CONN_TARGET))
+	{
+		/* errors have already been logged */
+		return false;
+	}
+
+	char *sql = "SELECT timescaledb_post_restore()";
```

Comment:
This doesn't work when the timescaledb extension is installed on non "public" schema. You may have factoring the schema at which timescaledb is installed.

e.g.
```
create extension timescaledb with schema "foo" version "2.13.0";
```

Reply:
done